connectors/sources/confluence.py

```diff
@@ -557,25 +616,42 @@ async def get_docs(self, filtering=None):
         Yields:
             dictionary: dictionary containing meta-data of the content.
         """
-        if self.spaces == ["*"]:
-            configured_spaces_query = "cql=type="
+        if filtering and filtering.has_advanced_rules():
+            advanced_rules = filtering.get_advanced_rules()
+            for query_info in advanced_rules:
+                query = query_info.get("query")
+                logger.debug(f"Fetching content using custom query: {query}")
+                async for document, download_link in self.search_by_query(query):
+                    if download_link:
+                        yield document, partial(
+                            self.download_attachment,
+                            download_link[1:],
+                            copy(document),
+                        )
+                    else:
+                        yield document, None
+
         else:
-            quoted_spaces = "','".join(self.spaces)
-            configured_spaces_query = f"cql=space in ('{quoted_spaces}') AND type="
-        await self.fetchers.put(self._space_coro)
-        await self.fetchers.put(
-            partial(
-                self._page_blog_coro,
-                f"{configured_spaces_query}blogpost&{CONTENT_QUERY}",
+            if self.spaces == ["*"]:
```

Comment:
```suggestion
            if self.spaces == [WILDCARD]:
```

suggestion: I think we should use a constant here as `*` could mean anything